tests/Unit/Framework/ActionTesting.hpp

```diff
@@ -408,35 +407,35 @@ class MockArrayElementProxy {
 
   template <typename Action, typename... Args>
   void simple_action(std::tuple<Args...> args) noexcept {
-    local_algorithm_.template simple_action<Action>(std::move(args));
+    mock_chare_element_.template simple_action<Action>(std::move(args));
   }
 
   template <typename Action>
   void simple_action() noexcept {
-    local_algorithm_.template simple_action<Action>();
+    mock_chare_element_.template simple_action<Action>();
   }
 
   template <typename Action, typename... Args>
   void threaded_action(std::tuple<Args...> args) noexcept {
-    local_algorithm_.template threaded_action<Action>(std::move(args));
+    mock_chare_element_.template threaded_action<Action>(std::move(args));
   }
 
   template <typename Action>
   void threaded_action() noexcept {
-    local_algorithm_.template threaded_action<Action>();
+    mock_chare_element_.template threaded_action<Action>();
   }
 
-  void set_terminate(bool t) noexcept { local_algorithm_.set_terminate(t); }
+  void set_terminate(bool t) noexcept { mock_chare_element_.set_terminate(t); }
 
   // Actions may call this, but since tests step through actions manually it has
   // no effect.
   void perform_algorithm() noexcept {}
   void perform_algorithm(const bool /*restart_if_terminated*/) noexcept {}
 
-  MockDistributedObject<Component>* ckLocal() { return &local_algorithm_; }
+  MockChareElement<Component>* ckLocal() { return &mock_chare_element_; }
```

Comment:
I find that I have no clue what a `MockChareElement` is. A chare element isn't even a concept in Charm++ so that seems like a bad name. Why is `MockDistributedObject` bad? This is exactly what it is.

Reply:
I'm happy to discuss better names. 

`MockChareElement` is the thing referred to by a `MockArrayElementProxy`, i.e. a single element of an Array chare, Group chare, or NodeGroup chare.  `MockChareElement` does not represent an entire chare (even for singletons which we treat as an Array with one element). I added 'Element' to the name `MockChareElement` to match 'Element' in the name `MockArrayElementProxy`.  (As opposed to `MockProxy` which is a proxy to an entire chare).

I am also considering renaming `MockArrayElementProxy` -> `MockChareElementProxy` to get the names further in sync,
and reflect the fact that we will soon have Groups and NodeGroups too.

I find the name `MockDistributedObject` very confusing, because the object is not "distributed": it is one element of an array chare that lives on a single core somewhere.   When I see the name `MockDistributedObject` I think of an entire array chare, which is distributed over many cores/nodes (like the thing that `MockProxy` refers to, which is an unordered_map of `MockChareElement`s aka `MockDistributedObject`s that is indexed by an `array_index`). Much of my confusion in ActionTesting came from the name `MockDistributedObject` (and that objects of type `MockDistributedObject` were named `local_algorithm`, and I didn't understand how something can be both distributed and local).

Reply:
Agreed at the SpECTRE call Dec 9 to keep it MockDistributedObject, and to eventually rename AlgorithmImpl -> DistributedObject.  "Distributed Object" is a regular term in HPC meaning an object that can be called remotely.